src/test/java/org/cactoos/http/HtWireTest.java

```diff
@@ -69,30 +76,71 @@ public void guessesCorrectPortForHttps() throws Exception {
         this.checkPorts("https://localhost", HtWireTest.HTTPS_PORT);
     }
 
+    @Test
+    public void guessesCorrectPortForExplicit() throws Exception {
+        final int port = 1234;
+        this.checkPorts(
+            new FormattedText("https://localhost:%d", port).asString(),
+            port
+        );
+    }
+
     @Test
     public void worksWithProvidedHostNameAndPort() throws IOException {
         new FtRemote(new TkText("Hello")).exec(
             home -> MatcherAssert.assertThat(
                 new TextOf(
                     new HtResponse(
                         new HtWire(home.getHost(), home.getPort()),
-                        new InputOf(
-                            new JoinedText(
-                                new TextOf("\r\n"),
-                                new TextOf("GET / HTTP/1.1"),
-                                new FormattedText(
-                                    "Host:%s",
-                                    home.getHost()
-                                )
-                            )
-                        )
+                        new GetInput(home)
                     )
                 ),
                 new TextHasString("HTTP/1.1 200")
             )
         );
     }
 
+    // @todo #63:30min HtWire should not close the socket opened to the remote
+    //  service (this includes not closing associated input and output
+    //  streams). It should instead return an `Input` with the socket's
+    //  inputstream "unread". Refactor accordingly and unignore this test.
+    @Ignore("see todo above")
+    @Test
+    @SuppressWarnings("PMD.AvoidCatchingGenericException")
+    public void closesSocketOnlyAfterResponseIsClosed() throws Exception {
+        new FtRemote(new TkText("Hey")).exec(
+            home -> {
+                @SuppressWarnings("resource")
+                final Socket socket = new Socket(
+                    home.getHost(),
+                    home.getPort()
+                );
+                try (InputStream ins = new HtWire(() -> socket).send(
+                    new GetInput(home)
+                ).stream()) {
+                    new Assertion<>(
+                        "must have a response",
+                        () -> ins.available(),
+                        Matchers.greaterThan(0)
+                    ).affirm();
+                    new Assertion<>(
+                        "must keep the socket open until response is closed",
+                        socket::isClosed,
+                        new IsNot<>(new IsTrue())
+                    ).affirm();
+                    // @checkstyle IllegalCatchCheck (1 line)
+                } catch (final Exception ex) {
+                    throw new IOException(ex);
+                }
+                new Assertion<>(
+                    "must close the socket once input response is closed",
```

Comment:
@victornoel I think at this point you're only testing the automatic close of try-with-resources, right?

Reply:
@llorllale well, I am testing that the `Socket` is closed once the `InputStream` was closed. I'm not testing that the `InputStream` was closed, I'm closing it via a try-with-resource: so I'm not testing the try-with-resource, I'm using it.

Reply:
@victornoel hmm closing a Socket's inputstream closes the socket itself. But, since we're dealing with a layer above `Socket#getInputstream()`, I guess we could allow it...

Reply:
@llorllale  I'm not sure I understand: the original ticket was to ensure the `Socket` (and so the `InputStream` and `OutpuStream`, since they are all altogether tied for closing) stays open after `send()` is called on `HtWire`. So the test is there for that. The fact that closing the `InputStream` is closing the socket is part of `Socket` contract, not `HtWire`.

Reply:
@llorllale I can actually remove this assertion, and the test will still be correct, but I thought it would be better to also test this :)